lib/scoring.rb

```diff
@@ -1,9 +1,39 @@
 module Scrabble
   class Scoring
+    CHART = {"A" => 1, "B" => 3, "C" => 3, "D" => 2, "E" => 1, "F" =>4, "G" => 2, "H" => 4, "I" => 1, "J" => 8, "K" => 5, "L" => 1, "M" => 3, "N" => 1, "O" => 1, "P" => 3, "Q" => 10, "R" => 1, "S" => 1, "T" => 1, "U" => 1, "V" => 4, "W" => 4, "X" => 8, "Y" => 4, "Z" => 10}
     def self.score(word)
-    end
+      return nil if word !~ /^[a-zA-Z]{0,7}$/ || word == ""
+      total = 0
+      # letters is a array of upcase letters
+      letters = word.upcase.chars.to_a
+
+      letters.map{ | letter |
```

Comment:
You don't need `map` here, `map` is used to convert an array to something else.  If you wanted to use `map` you could:
```ruby
scores = letters.map do |letter|
   CHART[letter.toupcase]
end
return scores.sum
```